Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -320,12 +321,28 @@ public OrderTicket MarketOnOpenOrder(Symbol symbol, int quantity, string tag = "
         [DocumentationAttribute(TradingAndOrders)]
         public OrderTicket MarketOnOpenOrder(Symbol symbol, decimal quantity, string tag = "", IOrderProperties orderProperties = null)
         {
+            var properties = orderProperties ?? DefaultOrderProperties?.Clone();
+            InvalidateGoodTilDateTimeInForce(properties);
+
             var security = Securities[symbol];
-            var request = CreateSubmitOrderRequest(OrderType.MarketOnOpen, security, quantity, tag, orderProperties ?? DefaultOrderProperties?.Clone());
+            var request = CreateSubmitOrderRequest(OrderType.MarketOnOpen, security, quantity, tag, properties);
 
             return SubmitOrderRequest(request);
         }
 
+        /// <summary>
+        /// Resets the time-in-force to the default <see cref="TimeInForce.GoodTilCanceled" /> if the given one is a <see cref="GoodTilDateTimeInForce"/>.
+        /// This is required for MOO and MOC orders, for which GTD is not supported.
+        /// </summary>
+        private static void InvalidateGoodTilDateTimeInForce(IOrderProperties orderProperties)
```

Comment:
Nice! Suggest let's add a one time warning message to the user about this. We should move the method to the end of the file with the other privates 👍